.pre-commit-config.yaml

```diff
@@ -0,0 +1,17 @@
+repos:
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: "v2.6.2"
+    hooks:
+      - id: prettier
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    rev: "v8.16.0"
+    hooks:
+      - id: eslint
+        files: \.[jt]sx?$ # *.js, *.jsx, *.ts and *.tsx
+        types: [file]
+        additional_dependencies:
+          - "eslint@8.16.0"
+          - "eslint-config-prettier@8.5.0"
+          - "prettier@2.6.2"
+          - "@typescript-eslint/eslint-plugin@5.27.0"
```

Comment:
Will this work for all files or just `staged`?

Reply:
Replying to your question: just `staged`.

But, since I'm proposing to use `Husky` instead of `pre-commit` tool, now all files are checked when `git commit` and `git push` are performed.